src/translators/crate.py

```diff
@@ -146,6 +146,93 @@ def _store_medatata(self, table_name, persisted_metadata):
         if (major <= 3):
             stmt = "insert into {} (table_name, entity_attrs) values (?,?) " \
                    "on duplicate key update entity_attrs = values(entity_attrs)"
+        else:
+            stmt = "insert into {} (table_name, entity_attrs) values (?,?) " \
+                   "on conflict(table_name) DO UPDATE SET entity_attrs = excluded.entity_attrs"
+            stmt = stmt.format(METADATA_TABLE_NAME)
+            self.cursor.execute(stmt, (table_name, persisted_metadata))
+
+
+    def _get_et_table_names(self, fiware_service=None):
+        """
+        Return the names of all the tables representing entity types.
+        :return: list(unicode)
+        """
+        op = "select distinct table_name from {} ".format(METADATA_TABLE_NAME)
+        if fiware_service:
+            where = "where table_name ~* '\"{}{}\"[.].*'"
+            op += where.format(TENANT_PREFIX, fiware_service.lower())
+
+        try:
+            self.cursor.execute(op)
+        except exceptions.ProgrammingError as e:
+            # Metadata table still not created
+            msg = "Could not retrieve METADATA_TABLE. Empty database maybe?. {}"
+            logging.debug(msg.format(e))
+            return []
+        return [r[0] for r in self.cursor.rows]
+
+
+    def _get_select_clause(self, attr_names, aggr_method, aggr_period):
+        if not attr_names:
+            return '*'
+
+        attrs = ['entity_type', 'entity_id']
+        if aggr_method:
+            if aggr_period:
+                attrs.append(
+                    "DATE_TRUNC('{}',{}) as {}".format(
+                        aggr_period, self.TIME_INDEX_NAME, self.TIME_INDEX_NAME)
+                )
+            # TODO: https://github.com/smartsdk/ngsi-timeseries-api/issues/106
+            m = '{}("{}") as "{}"'
+            attrs.extend(m.format(aggr_method, a, a) for a in set(attr_names))
+
+        else:
+            attrs.append(self.TIME_INDEX_NAME)
+            attrs.extend('"{}"'.format(a) for a in attr_names)
+
+        select = ','.join(attrs)
+        return select
+
+
+    def _get_limit(self, limit, last_n,env: dict = os.environ):
```

Comment:
We need to move the lookup of the default limit from `crate.py` to `sql_translator.py`, i.e. this code should be in the `_get_limit` of `sql_translator.py` whereas `crate.py` should have no `_get_limit`?